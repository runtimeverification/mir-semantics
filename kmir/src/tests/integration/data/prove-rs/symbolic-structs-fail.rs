#[allow(dead_code)]
struct IntStruct{
    field1: i16,
    field2: u8,
    field3: bool,
}

#[allow(dead_code)]
struct OtherStruct {
    field1: i16,
    field2: u8,
}

#[allow(dead_code)]
#[allow(unused)]
fn eats_struct_args(
    x1: i16,
    mut x2: IntStruct,
    x3: &OtherStruct,
) -> () {
    if (x2.field1 == (*x3).field1) {
        x2.field2 = (*x3).field2;
        x2.field3 = true;
    } else {
        x2.field1 = x1;
        x2.field3 = false;
    }

    assert!(x2.field3 || x2.field2 != (*x3).field2);
}

fn main () {
    let a = IntStruct{ field1: 0, field2: 1, field3: false};
    let b = OtherStruct{field1: 2, field2: 3};
    eats_struct_args(5, a, &b)
}
