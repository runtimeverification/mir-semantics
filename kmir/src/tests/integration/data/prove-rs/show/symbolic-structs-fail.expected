
┌─ 1 (root, init)
│   #execTerminator ( terminator ( ... kind: terminatorKindCall ( ... func: operandC
│   span: 0
│
│  (50 steps)
├─ 3 (split)
│   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 2 ) )
┃
┃ (branch)
┣━━┓ subst: .Subst
┃  ┃ constraint:
┃  ┃     ARG_INT2:Int ==Int ARG_INT5:Int
┃  │
┃  ├─ 4
┃  │   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 2 ) )
┃  │
┃  │  (56 steps)
┃  ├─ 6 (terminal)
┃  │   #EndProgram ~> .K
┃  │
┃  ┊  constraint: true
┃  ┊  subst: ...
┃  └─ 2 (leaf, target, terminal)
┃      #EndProgram ~> .K
┃
┗━━┓ subst: .Subst
   ┃ constraint:
   ┃     notBool ARG_INT2:Int ==Int ARG_INT5:Int
   │
   ├─ 5
   │   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 2 ) )
   │
   │  (81 steps)
   ├─ 7 (split)
   │   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 5 ) )
   ┃
   ┃ (branch)
   ┣━━┓ subst: .Subst
   ┃  ┃ constraint:
   ┃  ┃     notBool ARG_UINT3:Int ==Int ARG_UINT6:Int
   ┃  │
   ┃  ├─ 8
   ┃  │   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 5 ) )
   ┃  │
   ┃  │  (5 steps)
   ┃  ├─ 10 (terminal)
   ┃  │   #EndProgram ~> .K
   ┃  │
   ┃  ┊  constraint: true
   ┃  ┊  subst: ...
   ┃  └─ 2 (leaf, target, terminal)
   ┃      #EndProgram ~> .K
   ┃
   ┗━━┓ subst: .Subst
      ┃ constraint:
      ┃     ARG_UINT3:Int ==Int ARG_UINT6:Int
      │
      ├─ 9
      │   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 5 ) )
      │
      │  (4 steps)
      └─ 11 (stuck, leaf)
          #execTerminator ( terminator ( ... kind: terminatorKindCall ( ... func: operandC
          span: 65



